src/inference/include/openvino/runtime/intel_gna/gna_properties.hpp

```diff
@@ -0,0 +1,212 @@
+    std::string str;
+    is >> str;
+    if (str == "GNA_AUTO") {
+        execution_mode = ExecutionMode::AUTO;
+    } else if (str == "GNA_HW") {
+        execution_mode = ExecutionMode::HW;
+    } else if (str == "GNA_HW_WITH_SW_FBACK") {
+        execution_mode = ExecutionMode::HW_WITH_SW_FBACK;
+    } else if (str == "GNA_SW_EXACT") {
+        execution_mode = ExecutionMode::SW_EXACT;
+    } else if (str == "GNA_SW_FP32") {
+        execution_mode = ExecutionMode::SW_FP32;
+    } else {
+        throw ov::Exception{"Unsupported execution mode: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief Enum to define compile and execution targets
+ */
+enum class TargetDevice {
+    UNKNOWN = 0,
+    GNA_2_0 = 1,
+    GNA_3_0 = 2,
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const TargetDevice& target_device) {
+    switch (target_device) {
+    case TargetDevice::UNKNOWN:
+        return os << "";
+    case TargetDevice::GNA_2_0:
+        return os << "GNA_TARGET_2_0";
+    case TargetDevice::GNA_3_0:
+        return os << "GNA_TARGET_3_0";
+    default:
+        throw ov::Exception{"Unsupported target device!"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, TargetDevice& target_device) {
+    std::string str;
+    is >> str;
+    if (str.empty()) {
+        target_device = TargetDevice::UNKNOWN;
+    } else if (str == "GNA_TARGET_2_0") {
+        target_device = TargetDevice::GNA_2_0;
+    } else if (str == "GNA_TARGET_3_0") {
+        target_device = TargetDevice::GNA_3_0;
+    } else {
+        throw ov::Exception{"Unsupported target device: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief GNA proc_type setting that should be one of AUTO, HW, GNA_HW_WITH_SW_FBACK,
+ *        GNA_SW_EXACT or SW_FP32
+ */
+static constexpr Property<ExecutionMode> execution_mode{"GNA_DEVICE_MODE"};
+
+/**
+ * @brief The option to override the GNA HW execution target. May be one of TARGET_2_0, TARGET_3_0.
+ * By default (in case of no value set) the behavior depends on GNA HW availability:
+ * If GNA HW is present, use the option corresponding to this HW.
+ * If HW is not present, use the option corresponding to the latest fully supported GNA HW generation.
+ * A fully supported GNA HW generation means it must be supported by both the OV GNA Plugin and the core GNA Library.
+ * Currently, the latest supported GNA HW generation corresponds to GNA_TARGET_3_0.
+ */
+static constexpr Property<TargetDevice> execution_target{"GNA_EXEC_TARGET"};
+
+/**
+ * @brief The option to override the GNA HW compile target. May be one of TARGET_2_0, TARGET_3_0.
+ * By default the same as execution_target.
+ */
+static constexpr Property<TargetDevice> compile_target{"GNA_COMPILE_TARGET"};
+
+/**
+ * @brief if enabled produced minimum memory footprint for loaded network in GNA memory, default value is YES
+ */
+static constexpr Property<bool> memory_reuse{"GNA_COMPACT_MODE"};
```

Comment:
@dorloff do we need this option? from the description it seems that this option must always be `true`, because as a user - I don't understand why I need to use memory :) is it a WA for some issues?

One of the options - not to introduce the option in the new API, until you fix some issues users can use old option from old API.

Reply:
@ilya-lavrenov, I agree that it can be removed at some point, but at this stage the feature makes sense because we can ask a user to switch off this mode if something is wrong for their model - note that some of our customers are never sharing their full models. So I would choose to introduce it in the new API and deprecate as usual (probably transferring it into a compilation option).